src/main/java/com/library/model/entity/mongo/Book.java

```diff
@@ -0,0 +1,33 @@
+package com.library.model.entity.mongo;
+
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
+import com.library.model.dto.AuthorNameDto;
+import com.library.model.dto.PublisherTitleDto;
+import lombok.Getter;
+import lombok.Setter;
+import org.bson.types.ObjectId;
+import org.springframework.data.mongodb.core.mapping.Document;
+import org.springframework.data.mongodb.core.mapping.MongoId;
+
+import java.util.List;
+
+@Document
+@Getter
+@Setter
+public class Book {
+
+    @MongoId
+    @JsonSerialize(using= ToStringSerializer.class)
+    private ObjectId id;
+
+    private String title;
+
+    private Short publicationYear;
+
+    private Bookshelf bookshelf;
+
+    private PublisherTitleDto publisher;
```

Comment:
DTO's in document.... it looks strange.

Please, explain this decision.

Reply:
Should I create new object that embedded to Book document and map to DTOs?

Reply:
fixed in "fixed embedded fields in BookStored" commit

Reply:
Next time just "fixed" will be enough.